hazelcast/src/main/java/com/hazelcast/jet/JetService.java

```diff
@@ -183,15 +192,21 @@ default Job newLightJob(@Nonnull Pipeline p) {
      * It substantially reduces the overhead for jobs that take milliseconds to
      * complete.
      * <p>
-     * A light job will not be cancelled if the client disconnects. It's
+     * A light job will not be cancelled if the client disconnects. Its
      * potential failure will be only logged in member logs.
+     * <p>
+     * If pipeline and config instances will not mutate during the execution the
+     * immutableDefinitionAndConfig may be set to true. With that guarantee we can skip
+     * cloning of those parameters which also reduces overhead. This flag has an
+     * effect only in submission on member. In submission from client there is no
+     * way to skip the serialization part.
      */
-    Job newLightJob(@Nonnull Pipeline p, @Nonnull JobConfig config);
+    Job newLightJob(@Nonnull Pipeline p, @Nonnull JobConfig config, boolean immutableDefinitionAndConfig);
```

Comment:
I don't like the ```immutableDefinitionAndConfig``` name, if you have a better name, don't hesitate and let me know